openedx/core/lib/token_utils.py

```diff
@@ -1,124 +1,31 @@
-"""Utilities for working with ID tokens."""
-import json
-from time import time
-
-from django.conf import settings
-from django.utils.functional import cached_property
-from jwkest import jwk
-from jwkest.jws import JWS
-
-from edx_django_utils.monitoring import set_custom_metric
-from student.models import UserProfile, anonymous_id_for_user
+"""
+Deprecated JwtBuilder class for backward compatibility until edx-enterprise is updated.
+"""
+from openedx.core.djangoapps.oauth_dispatch.jwt import create_api_client_jwt
 
 
 class JwtBuilder(object):
-    """Utility for building JWTs.
-
-    Unifies diverse approaches to JWT creation in a single class. This utility defaults to using the system's
-    JWT configuration.
-
-    NOTE: This utility class will allow you to override the signing key and audience claim to support those
-    clients which still require this. This approach to JWT creation is DEPRECATED. Avoid doing this for new clients.
-
-    Arguments:
-        user (User): User for which to generate the JWT.
-
-    Keyword Arguments:
-        asymmetric (Boolean): Whether the JWT should be signed with this app's private key.
-        secret (string): Overrides configured JWT secret (signing) key. Unused if an asymmetric signature is requested.
-        issuer (string): Overrides configured JWT issuer.
     """
-
-    def __init__(self, user, asymmetric=False, secret=None, issuer=None):
+    Deprecated. Use openedx.core.djangoapps.oauth_dispatch.jwt.JwtBuilder directly.
+    """
+    def __init__(self, user, secret=None):
         self.user = user
-        self.asymmetric = asymmetric
         self.secret = secret
-        self.issuer = issuer
-        self.jwt_auth = settings.JWT_AUTH
-
-    def build_token(self, scopes, expires_in=None, aud=None, additional_claims=None):
-        """Returns a JWT access token.
-
-        Arguments:
-            scopes (list): Scopes controlling which optional claims are included in the token.
 
-        Keyword Arguments:
-            expires_in (int): Time to token expiry, specified in seconds.
-            aud (string): Overrides configured JWT audience claim.
-            additional_claims (dict): Additional claims to include in the token.
-
-        Returns:
-            str: Encoded JWT
+    def build_token(
+        self,
+        scopes=None,  # pylint: disable=unused-argument
+        expires_in=None,  # pylint: disable=unused-argument
+        aud=None,
+        additional_claims=None,
+    ):
         """
-        now = int(time())
-        expires_in = expires_in or self.jwt_auth['JWT_EXPIRATION']
-        set_custom_metric('jwt_expires_in', expires_in)
-
-        payload = {
-            # TODO Consider getting rid of this claim since we don't use it.
-            'aud': aud if aud else self.jwt_auth['JWT_AUDIENCE'],
-            'exp': now + expires_in,
-            'iat': now,
-            'iss': self.issuer if self.issuer else self.jwt_auth['JWT_ISSUER'],
```

Comment:
Please confirm: I imagine this is going away because a) we never override the issuer, and b) we never want to.

Reply:
yes.  :)